### DISCUSSION TITLE
Studio: Error filtering jsonb[] column (Malformed array literal)

### DISCUSSION THREAD

**Discussion by probablykasper** [supabase/supabase#16945] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Error when filtering `jsonb[]` columns.

In the devtools it shows POST requests to
- `https://api.supabase.io/platform/pg-meta/<PROJECT_ID>/query?key=public-pages` and
- `https://api.supabase.io/platform/pg-meta/<PROJECT_ID>/query?key=public-pages-count`
with the following response:
```js
{"message":"Failed to run sql query: malformed array literal: \"[]\""}
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a `jsonb[]` column with some records to filter for
2. Try filtering the column
3. Result: No rows ever show up, and an error message says `Error fetching records count. Please refresh the page.`

## Expected behavior

It should succeed and show the records

## Screenshots

<img width="582" alt="image" src="https://user-images.githubusercontent.com/11315492/232857597-d156149d-4cdc-43b5-8dab-36bd5afb95ca.png">

## System information

- OS: macOS
- Browser (if applies) Brave
- Version of supabase-js: N/A
- Version of Node.js: N/A

**Comment by GaryAustin1** [comment]
I can confirm it silently errors as described.
Also to get empty array `{}` postgres array syntax does work.

**Comment by Pictor13** [comment]
I also have the problem.

All POST requests to `https://api.supabase.io/platform/pg-meta/my-rproject/query?key=public-my_table` fail, when setting a `like` filter on a JSONB column.
Is a bit annoying because makes hard filtering rows of interest while debugging.

Is there any plan for a fix soon or later?

**Comment by doaboa** [comment]
Confirming this issue still exists. I tried switching my jsonb[] column to text[] as the supabase docs suggest, still receiving Malformed array literal.

Any planned fix?

**Reply by mariodev99** [reply]
any solution?

**Reply by GaryAustin1** [reply]
The issue was transferred to a discussion by someone at Supabase back in August of 23.  I doubt it is on their radar or they believe it is solved.
